test/test_autograd.py

```diff
@@ -3227,6 +3227,8 @@ def test_profiler_shapes(self):
         last_end = 0
 
         for event in prof.function_events:
+            if event.name == 'aten::linear':
```

Comment:
This looks very arbitrary? Why do you need to special case linear here?
If there is a good reason, it would help to have a comment here explaining it.

Reply:
The test here is somewhat hard-coded. `top_level_expected_events_and_shapes` maps top level profiled kernel.

Right now since we expose `aten::linear`, the old hard-coded names / shapes are not top level any more. So we are skipping `aten::linear`. I'll put a note here.

Reply:
yea, this looks a little funky to me too, but i dont really know anything about it. cc @ilia-cher

Reply:
this is very hacky and arbitrary, please make sure to fix the test the right way

what this test does: runs profiler, gets a list of function (operator) events - note that some ops are called by other ops, then we go through the events and look at the ones that don't have a parent (that is they are the top level events)

your PR as i understand changes the ops and thus changes the output of the profiler